Runtime error using NATS client without go ARI package

Hi! 

Thank for this great project.

I'm connecting into the proxy using a python client and can manage to exchange a few messages successfully. 
I'm getting this error after sending a ChannelRing to a channel:

> "panic: runtime error: invalid memory address or nil pointer dereference" error

The ChannelRing is executed successfully, but this error appear right after and the proxy process crashes.

`
ari-proxy_1  | t=2020-03-30T18:55:14+0000 lvl=info msg="starting ari-proxy server" version=5.0.1
ari-proxy_1  | t=2020-03-30T18:55:18+0000 lvl=warn msg="failed to publish NATS message" subject= data="&{Error: Data:<nil> Key:<nil> Keys:[]}" error="nats: invalid subject"
ari-proxy_1  | panic: runtime error: invalid memory address or nil pointer dereference
ari-proxy_1  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x867b53]
ari-proxy_1  |
ari-proxy_1  | goroutine 45 [running]:
ari-proxy_1  | github.com/CyCoreSystems/ari-proxy/v5/server.(*Server).dialogsForEvent(0xc00009c500, 0xb77140, 0xc0001be100, 0xa3cfc0, 0xc0001be100, 0xc0000266f0)
ari-proxy_1  | 	/home/travis/gopath/src/github.com/CyCoreSystems/ari-proxy/server/events.go:7 +0x113
ari-proxy_1  | github.com/CyCoreSystems/ari-proxy/v5/server.(*Server).runEventHandler(0xc00009c500, 0xb72a40, 0xc00005fb80)
ari-proxy_1  | 	/home/travis/gopath/src/github.com/CyCoreSystems/ari-proxy/server/server.go:276 +0x51d
ari-proxy_1  | created by github.com/CyCoreSystems/ari-proxy/v5/server.(*Server).listen
ari-proxy_1  | 	/home/travis/gopath/src/github.com/CyCoreSystems/ari-proxy/server/server.go:221 +0x138e
vagrant_ari-proxy_1 exited with code 2
`

```
INFO[03-30|19:56:53] verbose logging enabled
INFO[03-30|19:56:53] starting ari-proxy server                version=master
DBUG[03-30|19:56:53] starting listener
DBUG[03-30|19:56:53] listening for events                     application=endpoint-route
DBUG[03-30|19:57:03] event received                           kind=ChannelVarset
DBUG[03-30|19:57:03] listening for events                     application=endpoint-route
DBUG[03-30|19:57:03] event received                           kind=ChannelCreated
DBUG[03-30|19:57:03] listening for events                     application=endpoint-route
DBUG[03-30|19:57:03] event received                           kind=ChannelDialplan
DBUG[03-30|19:57:03] listening for events                     application=endpoint-route
DBUG[03-30|19:57:03] event received                           kind=ChannelVarset
DBUG[03-30|19:57:03] listening for events                     application=endpoint-route
DBUG[03-30|19:57:03] event received                           kind=StasisStart
DBUG[03-30|19:57:03] listening for events                     application=endpoint-route
DBUG[03-30|19:57:03] received request                         kind=ChannelRing
WARN[03-30|19:57:03] failed to publish NATS message           subject= data="&{Error: Data:<nil> Key:<nil> Keys:[]}" error="nats: invalid subject"
DBUG[03-30|19:57:03] event received                           kind=DeviceStateChanged
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x876d03]

goroutine 42 [running]:
github.com/CyCoreSystems/ari-proxy/v5/server.(*Server).dialogsForEvent(0xc000020600, 0xb855e0, 0xc000021680, 0xa4fbc0, 0xc000021680, 0xc00002e7e0)
	/usr/src/ari-proxy/server/events.go:7 +0x113
github.com/CyCoreSystems/ari-proxy/v5/server.(*Server).runEventHandler(0xc000020600, 0xb807e0, 0xc000063c80)
	/usr/src/ari-proxy/server/server.go:276 +0x518
created by github.com/CyCoreSystems/ari-proxy/v5/server.(*Server).listen
	/usr/src/ari-proxy/server/server.go:221 +0x1275
```

Yes, looks like the problem is with DeviceStateChanged event.

Here is a log after hangup a channel which is already active before starting the proxy:

```
INFO[03-30|20:03:13] verbose logging enabled
INFO[03-30|20:03:13] starting ari-proxy server                version=master
DBUG[03-30|20:03:13] starting listener
DBUG[03-30|20:03:13] listening for events                     application=endpoint-route
DBUG[03-30|20:03:19] event received                           kind=ChannelHangupRequest
DBUG[03-30|20:03:19] listening for events                     application=endpoint-route
DBUG[03-30|20:03:19] event received                           kind=StasisEnd
DBUG[03-30|20:03:19] listening for events                     application=endpoint-route
DBUG[03-30|20:03:19] event received                           kind=ChannelVarset
DBUG[03-30|20:03:19] listening for events                     application=endpoint-route
DBUG[03-30|20:03:19] event received                           kind=ChannelHangupRequest
DBUG[03-30|20:03:19] listening for events                     application=endpoint-route
DBUG[03-30|20:03:19] event received                           kind=ChannelDestroyed
DBUG[03-30|20:03:19] listening for events                     application=endpoint-route
DBUG[03-30|20:03:19] event received                           kind=DeviceStateChanged
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x876d03]

goroutine 42 [running]:
github.com/CyCoreSystems/ari-proxy/v5/server.(*Server).dialogsForEvent(0xc000020600, 0xb855e0, 0xc000021280, 0xa4fbc0, 0xc000021280, 0xc00002e750)
	/usr/src/ari-proxy/server/events.go:7 +0x113
github.com/CyCoreSystems/ari-proxy/v5/server.(*Server).runEventHandler(0xc000020600, 0xb807e0, 0xc000063c80)
	/usr/src/ari-proxy/server/server.go:276 +0x518
```

--- a/server/events.go
+++ b/server/events.go
@@ -4,6 +4,10 @@ import "github.com/CyCoreSystems/ari/v5"
 
 func (s *Server) dialogsForEvent(e ari.Event) (ret []string) {
 	for _, k := range e.Keys() {
+		if k == nil {
+			s.Log.Debug("received nil key for event", "event", e)
+			continue
+		}
 		ret = append(ret, s.Dialog.List(k.Kind, k.ID)...)
 	}
 	return

